pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1303,10 +1303,12 @@ def _handle_distribution_tree(declarative_content):
                     update_references_to_save.append(update_reference)
 
         if update_collection_to_save:
-            UpdateCollection.objects.bulk_create(update_collection_to_save)
+            UpdateCollection.objects.bulk_create(update_collection_to_save, ignore_conflicts=True)
 
         if update_collection_packages_to_save:
-            UpdateCollectionPackage.objects.bulk_create(update_collection_packages_to_save)
```

Comment:
UpdateCollectionPackage has no unique_together key defined

Reply:
If there's no uniqueness-constraint, then we're never going to have conflicts to ignore, yeah?

Reply:
Yeah - but depending on the exact mechanics of the sync though, we might be saving trash objects to the database in that event

Reply:
UpdateCollection is unique by (name, update_record_id). So if two threads are persisting the same UpdateRecord at the same time, they would collide on trying to create/persist the UpdatePackageCollection - I think only one would be created. I believe they might end up with duplicated UpdateReferences, because the references are only unique by-id. Does that feel correct?

Reply:
It is almost certainly possible to end up with extraneous records in the db, alas. However - saving UpdateRecords already has a certain amount of window-closing - two threads trying to persis the same updaterecord at the same time, whichever notices that it's "already there" will then go into the advisory-resolve codepath.

I gues, from my POV, that is a) a pretty tiny window, with b) not a huge impact (these records are teeny), and c) not what you're fixing here anyway. I'd drop the ignoreConflict= from the models where you can't actually **have** a conflict, and concentrate on whether using it on the remainder is going to cause problems on the way to committing this transaction.

Reply:
I agree we do not need ignore_conflicts here, but I'm not sure what is more confusing from the readability point of view. Having one bulk_create without ignore_conflicts or having a useless ignore_conflicts but consistent with the rest. Since it doesn't affect functionality maybe being consistent is less confusing.

I also noticed that we are not using objects returned from bulk_create, so at least this problem won't bite us down the road like it did with advisories in the migration plugin, @ggainey

Reply:
+1 to keep ignore_conflicts everywhere for consistency. In 6 months when looking at the code we gonna ask ourselves why here we have ignore_conflicts and there not.
pulp_container also uses it in sync stages

Reply:
Updated